server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1250,7 +1254,11 @@ private Delete(final StreamInput in) throws IOException {
                 id = uidObject.id();
             }
             this.version = in.readLong();
-            this.versionType = VersionType.fromValue(in.readByte());
+            if (format <= FORMAT_SEQ_NO) {
+                this.versionType = VersionType.fromValue(in.readByte());
+            } else {
+                this.versionType = VersionType.EXTERNAL; // versionType is removed in 7.0
```

Comment:
I wonder if it's worth while to back port the format identifier here .. at very least add a comment and assert that we will never use the format id of 7.0 here if we end up changing the format in the 6.x life time.